repentogon/LuaInterfaces/LuaItemPool.cpp

```diff
@@ -31,9 +43,11 @@ LUA_FUNCTION(Lua_ItemPoolGetCardEx)
 LUA_FUNCTION(Lua_ItemPoolGetCollectibleEx) {
 	ItemPool* itemPool = lua::GetUserdata<ItemPool*>(L, 1, lua::Metatables::ITEM_POOL, "ItemPool");
 	int poolType = (int)luaL_checkinteger(L, 2);
-	uint32_t seed = (unsigned int)luaL_optinteger(L, 3, Isaac::genrand_int32());
-	if (seed == 0) seed = 1;  // IDC
-	bool keepWeight = luaL_optinteger(L, 4, 1); //Fun Fact: main.lua creates a wrapper for GetCollectible that inverts the boolean value passed as Decrease, then passes it as an integer.
+	bool decrease = lua::luaL_optboolean(L, 3, false);
+	uint32_t seed = (unsigned int)luaL_optinteger(L, 4, Isaac::genrand_int32());
+	if (seed == 0) {
+		seed = GetValidRNGSeed();
```

Comment:
Please just use 1, the effect to the RNG here is negligible. It's not worth the extra complexity of anything involving a loop.

Reply:
Updated